app/src/main/java/org/mozilla/focus/ScreenController.kt

```diff
@@ -41,24 +41,9 @@ object ScreenController {
             searchTerms = urlStr.trim()
         }
 
-        // TODO: could this ever happen where browserFragment is on top? and do we need to do anything special for it?
-        val browserFragment = fragmentManager.findFragmentByTag(BrowserFragment.FRAGMENT_TAG)
-        val isSearch = !TextUtils.isEmpty(searchTerms)
-        if (browserFragment != null && browserFragment.isVisible) {
-            // Reuse existing visible fragment - in this case we know the user is already browsing.
-            // The fragment might exist if we "erased" a browsing session, hence we need to check
-            // for visibility in addition to existence.
-            (browserFragment as BrowserFragment).loadUrl(updatedUrlStr)
-
-            // And this fragment can be removed again.
-            fragmentManager.beginTransaction()
-                    .replace(R.id.container, browserFragment)
-                    .addToBackStack(null)
-                    .commit()
-        } else {
-            SessionManager.getInstance().createSession(Source.USER_ENTERED, updatedUrlStr)
-        }
+        showBrowserScreenForUrl(fragmentManager, updatedUrlStr)
 
+        val isSearch = !TextUtils.isEmpty(searchTerms)
```

Comment:
`isSearch` and `isUrl` are basically the same variable, maybe you can combine them.